### ISSUE TITLE
script is using pxc_cluster_view even with --active_failover=1

### ISSUE THREAD

**Issue by altmannmarcelo** [Tusamarco/proxy_sql_tools#11]
Hi Marco,

I have my env configure as follow:

```
node1 - 192.168.91.41
node2 - 192.168.91.42
node3 - 192.168.91.43
node4 - 192.168.91.44

SELECT * FROM scheduler;
+----+--------+-------------+-----------------------------------+--------------------------------------------------------------------------------------------------------------------------------------+------+------+------+------+---------+
| id | active | interval_ms | filename                          | arg1                                                                                                                                 | arg2 | arg3 | arg4 | arg5 | comment |
+----+--------+-------------+-----------------------------------+--------------------------------------------------------------------------------------------------------------------------------------+------+------+------+------+---------+
| 10 | 1      | 2000        | /var/lib/proxysql/galera_check.pl | -u=admin -p=admin -h=127.0.0.1 -H=50:W,51:R -P=6032 --main_segment=0 --debug=1 --active_failover=1 --log=/var/lib/proxysql/galeraLog | NULL | NULL | NULL | NULL |         |
+----+--------+-------------+-----------------------------------+--------------------------------------------------------------------------------------------------------------------------------------+------+------+------+------+---------+
1 row in set (0.01 sec)

```
```
select * from pxc_cluster_view order by SEGMENT,LOCAL_INDEX;
+-----------+--------------------------------------+--------+-------------+---------+
| HOST_NAME | UUID                                 | STATUS | LOCAL_INDEX | SEGMENT |
+-----------+--------------------------------------+--------+-------------+---------+
| node2     | 40ad5f1d-ca46-11e7-8c67-3395b610aec7 | SYNCED |           0 |       0 |
| node4     | 860e97cb-ca46-11e7-b5b6-cf8d675e4da5 | SYNCED |           1 |       0 |
| node1     | be422657-ca46-11e7-9c77-72eb711d1d07 | SYNCED |           2 |       0 |
| node3     | c32ae2c6-ca46-11e7-9801-9e954a370b9f | SYNCED |           3 |       0 |
+-----------+--------------------------------------+--------+-------------+---------+
4 rows in set (0.00 sec)
```
Based on cluster view, next node to be consider is node4. I have mysql_server configured as follow:
```
SELECT * FROM mysql_servers;
+--------------+---------------+------+--------+--------+-------------+-----------------+---------------------+---------+----------------+---------+
| hostgroup_id | hostname      | port | status | weight | compression | max_connections | max_replication_lag | use_ssl | max_latency_ms | comment |
+--------------+---------------+------+--------+--------+-------------+-----------------+---------------------+---------+----------------+---------+
| 8050         | 192.168.91.41 | 3306 | ONLINE | 10000  | 0           | 1000            | 0                   | 0       | 0              |         |
| 8050         | 192.168.91.42 | 3306 | ONLINE | 1000   | 0           | 1000            | 0                   | 0       | 0              |         |
| 8050         | 192.168.91.43 | 3306 | ONLINE | 100    | 0           | 1000            | 0                   | 0       | 0              |         |
| 8050         | 192.168.91.44 | 3306 | ONLINE | 1      | 0           | 1000            | 0                   | 0       | 0              |         |
| 51           | 192.168.91.41 | 3306 | ONLINE | 10000  | 0           | 1000            | 0                   | 0       | 0              |         |
| 51           | 192.168.91.42 | 3306 | ONLINE | 10000  | 0           | 1000            | 0                   | 0       | 0              |         |
| 51           | 192.168.91.43 | 3306 | ONLINE | 1      | 0           | 1000            | 0                   | 0       | 0              |         |
| 51           | 192.168.91.44 | 3306 | ONLINE | 1      | 0           | 1000            | 0                   | 0       | 0              |         |
| 50           | 192.168.91.42 | 3306 | ONLINE | 1000   | 0           | 1000            | 0                   | 0       | 0              |         |
+--------------+---------------+------+--------+--------+-------------+-----------------+---------------------+---------+----------------+---------+
9 rows in set (0.01 sec)
```

Based on the weight of 8000 hg, next node to be elected should be node1. But when I stop node2, node4 is configured as active for hg 50:

```
SELECT * FROM mysql_servers;
+--------------+---------------+------+--------+--------+-------------+-----------------+---------------------+---------+----------------+---------+
| hostgroup_id | hostname      | port | status | weight | compression | max_connections | max_replication_lag | use_ssl | max_latency_ms | comment |
+--------------+---------------+------+--------+--------+-------------+-----------------+---------------------+---------+----------------+---------+
| 8050         | 192.168.91.41 | 3306 | ONLINE | 10000  | 0           | 1000            | 0                   | 0       | 0              |         |
| 8050         | 192.168.91.42 | 3306 | ONLINE | 1000   | 0           | 1000            | 0                   | 0       | 0              |         |
| 8050         | 192.168.91.43 | 3306 | ONLINE | 100    | 0           | 1000            | 0                   | 0       | 0              |         |
| 8050         | 192.168.91.44 | 3306 | ONLINE | 1      | 0           | 1000            | 0                   | 0       | 0              |         |
| 51           | 192.168.91.41 | 3306 | ONLINE | 10000  | 0           | 1000            | 0                   | 0       | 0              |         |
| 51           | 192.168.91.42 | 3306 | ONLINE | 10000  | 0           | 1000            | 0                   | 0       | 0              |         |
| 51           | 192.168.91.43 | 3306 | ONLINE | 1      | 0           | 1000            | 0                   | 0       | 0              |         |
| 51           | 192.168.91.44 | 3306 | ONLINE | 1      | 0           | 1000            | 0                   | 0       | 0              |         |
| 50           | 192.168.91.44 | 3306 | ONLINE | 1      | 0           | 1000            | 0                   | 0       | 0              |         |
+--------------+---------------+------+--------+--------+-------------+-----------------+---------------------+---------+----------------+---------+
9 rows in set (0.00 sec)
```

**Comment by altmannmarcelo** [comment]
A new test with debug enabled and segment = 1

```
root@localhost (performance_schema) > select * from pxc_cluster_view order by SEGMENT,LOCAL_INDEX;
+-----------+--------------------------------------+--------+-------------+---------+
| HOST_NAME | UUID                                 | STATUS | LOCAL_INDEX | SEGMENT |
+-----------+--------------------------------------+--------+-------------+---------+
| node3     | 0348163f-ca4f-11e7-92a9-2217a3435765 | SYNCED |           0 |       1 |
| node4     | 0eb4a1fe-ca4f-11e7-93cc-935fc9c4628b | SYNCED |           1 |       1 |
| node2     | 134521cc-ca4f-11e7-8ed1-0eb8038826b7 | SYNCED |           2 |       1 |
| node1     | f08ea4b5-ca4e-11e7-8c0b-cae7b94ffd25 | SYNCED |           3 |       1 |
+-----------+--------------------------------------+--------+-------------+---------+
4 rows in set (0.01 sec)

SELECT * FROM scheduler;
+----+--------+-------------+-----------------------------------+--------------------------------------------------------------------------------------------------------------------------------------+------+------+------+------+---------+
| id | active | interval_ms | filename                          | arg1                                                                                                                                 | arg2 | arg3 | arg4 | arg5 | comment |
+----+--------+-------------+-----------------------------------+--------------------------------------------------------------------------------------------------------------------------------------+------+------+------+------+---------+
| 10 | 1      | 2000        | /var/lib/proxysql/galera_check.pl | -u=admin -p=admin -h=127.0.0.1 -H=50:W,51:R -P=6032 --main_segment=1 --debug=1 --active_failover=1 --log=/var/lib/proxysql/galeraLog | NULL | NULL | NULL | NULL |         |
+----+--------+-------------+-----------------------------------+--------------------------------------------------------------------------------------------------------------------------------------+------+------+------+------+---------+
1 row in set (0.00 sec)

SELECT * FROM mysql_servers;
+--------------+---------------+------+--------+--------+-------------+-----------------+---------------------+---------+----------------+---------+
| hostgroup_id | hostname      | port | status | weight | compression | max_connections | max_replication_lag | use_ssl | max_latency_ms | comment |
+--------------+---------------+------+--------+--------+-------------+-----------------+---------------------+---------+----------------+---------+
| 50           | 192.168.91.41 | 3306 | ONLINE | 10000  | 0           | 1000            | 0                   | 0       | 0              |         |
| 8050         | 192.168.91.41 | 3306 | ONLINE | 10000  | 0           | 1000            | 0                   | 0       | 0              |         |
| 8050         | 192.168.91.42 | 3306 | ONLINE | 1000   | 0           | 1000            | 0                   | 0       | 0              |         |
| 8050         | 192.168.91.43 | 3306 | ONLINE | 100    | 0           | 1000            | 0                   | 0       | 0              |         |
| 8050         | 192.168.91.44 | 3306 | ONLINE | 1      | 0           | 1000            | 0                   | 0       | 0              |         |
| 51           | 192.168.91.41 | 3306 | ONLINE | 10000  | 0           | 1000            | 0                   | 0       | 0              |         |
| 51           | 192.168.91.42 | 3306 | ONLINE | 10000  | 0           | 1000            | 0                   | 0       | 0              |         |
| 51           | 192.168.91.43 | 3306 | ONLINE | 1      | 0           | 1000            | 0                   | 0       | 0              |         |
| 51           | 192.168.91.44 | 3306 | ONLINE | 1      | 0           | 1000            | 0                   | 0       | 0              |         |
+--------------+---------------+------+--------+--------+-------------+-----------------+---------------------+---------+----------------+---------+
9 rows in set (0.01 sec)



-- After shutdown node1 (192.168.91.41)

SELECT * FROM mysql_servers;
+--------------+---------------+------+--------+--------+-------------+-----------------+---------------------+---------+----------------+---------+
| hostgroup_id | hostname      | port | status | weight | compression | max_connections | max_replication_lag | use_ssl | max_latency_ms | comment |
+--------------+---------------+------+--------+--------+-------------+-----------------+---------------------+---------+----------------+---------+
| 8050         | 192.168.91.41 | 3306 | ONLINE | 10000  | 0           | 1000            | 0                   | 0       | 0              |         |
| 8050         | 192.168.91.42 | 3306 | ONLINE | 1000   | 0           | 1000            | 0                   | 0       | 0              |         |
| 8050         | 192.168.91.43 | 3306 | ONLINE | 100    | 0           | 1000            | 0                   | 0       | 0              |         |
| 8050         | 192.168.91.44 | 3306 | ONLINE | 1      | 0           | 1000            | 0                   | 0       | 0              |         |
| 51           | 192.168.91.41 | 3306 | ONLINE | 10000  | 0           | 1000            | 0                   | 0       | 0              |         |
| 51           | 192.168.91.42 | 3306 | ONLINE | 10000  | 0           | 1000            | 0                   | 0       | 0              |         |
| 51           | 192.168.91.43 | 3306 | ONLINE | 1      | 0           | 1000            | 0                   | 0       | 0              |         |
| 51           | 192.168.91.44 | 3306 | ONLINE | 1      | 0           | 1000            | 0                   | 0       | 0              |         |
| 50           | 192.168.91.43 | 3306 | ONLINE | 100    | 0           | 1000            | 0                   | 0       | 0              |         |
+--------------+---------------+------+--------+--------+-------------+-----------------+---------------------+---------+----------------+---------+
9 rows in set (0.00 sec)
```

LOG:

```
2017/11/15 21:53:56.508:[INFO] START EXECUTION
2017/11/15 21:53:56.508:[INFO]  Inizializing hostgroup 50 wwith maintenance HG 8050 bw
2017/11/15 21:53:56.508:[INFO]  Inizializing hostgroup 51 rwith maintenance HG 9051 mr
2017/11/15 21:53:56.535:[INFO]  Connecting to ProxySQL DBI:mysql:host=127.0.0.1;port=6032
2017/11/15 21:53:56.535:[INFO]  Galera cluster object created
2017/11/15 21:53:56.538:[DEBUG] Calculate retry from comment Node:192.168.91.41 port:3306 hg:50 Time IN
2017/11/15 21:53:56.538:[DEBUG] Calculate retry from comment Node:192.168.91.41 port:3306 hg:50 Time OUT
2017/11/15 21:53:56.538:[INFO]  Galera cluster node   192.168.91.41:3306:HG=50
2017/11/15 21:53:56.538:[DEBUG] Calculate retry from comment Node:192.168.91.41 port:3306 hg:51 Time IN
2017/11/15 21:53:56.538:[DEBUG] Calculate retry from comment Node:192.168.91.41 port:3306 hg:51 Time OUT
2017/11/15 21:53:56.538:[INFO]  Galera cluster node   192.168.91.41:3306:HG=51
2017/11/15 21:53:56.538:[DEBUG] Calculate retry from comment Node:192.168.91.42 port:3306 hg:51 Time IN
2017/11/15 21:53:56.538:[DEBUG] Calculate retry from comment Node:192.168.91.42 port:3306 hg:51 Time OUT
2017/11/15 21:53:56.538:[INFO]  Galera cluster node   192.168.91.42:3306:HG=51
2017/11/15 21:53:56.538:[DEBUG] Calculate retry from comment Node:192.168.91.43 port:3306 hg:51 Time IN
2017/11/15 21:53:56.538:[DEBUG] Calculate retry from comment Node:192.168.91.43 port:3306 hg:51 Time OUT
2017/11/15 21:53:56.538:[INFO]  Galera cluster node   192.168.91.43:3306:HG=51
2017/11/15 21:53:56.538:[DEBUG] Calculate retry from comment Node:192.168.91.44 port:3306 hg:51 Time IN
2017/11/15 21:53:56.538:[DEBUG] Calculate retry from comment Node:192.168.91.44 port:3306 hg:51 Time OUT
2017/11/15 21:53:56.538:[INFO]  Galera cluster node   192.168.91.44:3306:HG=51
2017/11/15 21:53:56.538:[DEBUG] Calculate retry from comment Node:192.168.91.41 port:3306 hg:8050 Time IN
2017/11/15 21:53:56.539:[DEBUG] Calculate retry from comment Node:192.168.91.41 port:3306 hg:8050 Time OUT
2017/11/15 21:53:56.539:[INFO]  Galera cluster node   192.168.91.41:3306:HG=8050
2017/11/15 21:53:56.539:[DEBUG] Calculate retry from comment Node:192.168.91.42 port:3306 hg:8050 Time IN
2017/11/15 21:53:56.539:[DEBUG] Calculate retry from comment Node:192.168.91.42 port:3306 hg:8050 Time OUT
2017/11/15 21:53:56.539:[INFO]  Galera cluster node   192.168.91.42:3306:HG=8050
2017/11/15 21:53:56.539:[DEBUG] Calculate retry from comment Node:192.168.91.43 port:3306 hg:8050 Time IN
2017/11/15 21:53:56.539:[DEBUG] Calculate retry from comment Node:192.168.91.43 port:3306 hg:8050 Time OUT
2017/11/15 21:53:56.539:[INFO]  Galera cluster node   192.168.91.43:3306:HG=8050
2017/11/15 21:53:56.539:[DEBUG] Calculate retry from comment Node:192.168.91.44 port:3306 hg:8050 Time IN
2017/11/15 21:53:56.539:[DEBUG] Calculate retry from comment Node:192.168.91.44 port:3306 hg:8050 Time OUT
2017/11/15 21:53:56.539:[INFO]  Galera cluster node   192.168.91.44:3306:HG=8050
2017/11/15 21:53:56.539:[INFO]  Galera cluster nodes loaded
2017/11/15 21:53:56.539:[INFO]  Creating new thread to manage server check:192.168.91.41:3306:HG50
2017/11/15 21:53:56.556:[INFO]  Creating new thread to manage server check:192.168.91.41:3306:HG51
2017/11/15 21:53:56.570:[INFO]  Creating new thread to manage server check:192.168.91.42:3306:HG51
2017/11/15 21:53:56.583:[INFO]  Creating new thread to manage server check:192.168.91.43:3306:HG51
2017/11/15 21:53:56.597:[INFO]  Creating new thread to manage server check:192.168.91.44:3306:HG51
2017/11/15 21:53:56.583:[DEBUG]  Node check START 192.168.91.42:3306:HG51
2017/11/15 21:53:56.603:[DEBUG]  Node check END 192.168.91.42:3306:HG51
2017/11/15 21:53:56.614:[INFO]  Creating new thread to manage server check:192.168.91.41:3306:HG8050
2017/11/15 21:53:56.597:[DEBUG]  Node check START 192.168.91.43:3306:HG51
2017/11/15 21:53:56.615:[DEBUG]  Node check END 192.168.91.43:3306:HG51
2017/11/15 21:53:56.630:[INFO]  Creating new thread to manage server check:192.168.91.42:3306:HG8050
2017/11/15 21:53:56.614:[DEBUG]  Node check START 192.168.91.44:3306:HG51
2017/11/15 21:53:56.634:[DEBUG]  Node check END 192.168.91.44:3306:HG51
2017/11/15 21:53:56.644:[INFO]  Creating new thread to manage server check:192.168.91.43:3306:HG8050
2017/11/15 21:53:56.658:[INFO]  Creating new thread to manage server check:192.168.91.44:3306:HG8050
2017/11/15 21:53:56.644:[DEBUG]  Node check START 192.168.91.42:3306:HG8050
2017/11/15 21:53:56.666:[DEBUG]  Node check END 192.168.91.42:3306:HG8050
2017/11/15 21:53:56.658:[DEBUG]  Node check START 192.168.91.43:3306:HG8050
2017/11/15 21:53:56.677:[DEBUG]  Node check END 192.168.91.43:3306:HG8050
2017/11/15 21:53:56.674:[DEBUG]  Node check START 192.168.91.44:3306:HG8050
2017/11/15 21:53:56.695:[DEBUG]  Node check END 192.168.91.44:3306:HG8050
2017/11/15 21:53:56.688:[INFO]  Thread joined :   3
2017/11/15 21:53:56.703:[INFO]  Thread joined :   4
2017/11/15 21:53:56.714:[INFO]  Thread joined :   5
2017/11/15 21:53:56.724:[INFO]  Thread joined :   7
2017/11/15 21:53:56.735:[INFO]  Thread joined :   8
2017/11/15 21:53:56.744:[INFO]  Thread joined :   9
2017/11/15 21:53:56.839:[WARN] Check timeout Node ip : 192.168.91.41 ,  THID 1 (taken: 300.019025802612   max_allowed: 300  over for ms: 0.0190258026123047
12017/11/15 21:53:56.839:[WARN] Check timeout Node ip : 192.168.91.41 ,  THID 2 (taken: 300.019025802612   max_allowed: 300  over for ms: 0.0190258026123047
12017/11/15 21:53:56.839:[WARN] Check timeout Node ip : 192.168.91.41 ,  THID 6 (taken: 300.019025802612   max_allowed: 300  over for ms: 0.0190258026123047
12017/11/15 21:53:56.839:[INFO]  Multi Thread execution done in :   300.23193359375(ms)
2017/11/15 21:53:56.839:[INFO]  Evaluate nodes state
2017/11/15 21:53:56.840:[WARN] Fail-over in action Using Method = 1
2017/11/15 21:53:56.840:[INFO] Special Backup - Group found! I am electing a node to writer following the indications
 This Node Try to become the new WRITER for HG 50 Server details: 192.168.91.43:3306:HG8050
2017/11/15 21:53:56.840:[INFO] This Node Try to become a WRITER promoting to HG 50 192.168.91.43:3306:HG 8050
2017/11/15 21:53:56.854:[WARN]  Move node:192.168.91.43:33061001000 SQL:INSERT INTO mysql_servers (hostname,hostgroup_id,port,weight,max_connections) VALUES ('192.168.91.43',50,3306,100,1000);
2017/11/15 21:53:56.854:[WARN] !!!! FAILOVER !!!!!
 Cluster was without WRITER I have try to restore service promoting a node
2017/11/15 21:53:56.854:[INFO] END EXECUTION Total Time(ms):345.828056335449
```

**Comment by Tusamarco** [comment]
@altmannmarcelo Hi, first of all tks to try the script.
Now .. why the node was not elected? 
If you check the log you will see this:
```
2017/11/15 21:53:56.839:[WARN] Check timeout Node ip : 192.168.91.41 ,  THID 1 (taken: 300.019025802612   max_allowed: 300  over for ms: 0.0190258026123047
12017/11/15 21:53:56.839:[WARN] Check timeout Node ip : 192.168.91.41 ,  THID 2 (taken: 300.019025802612   max_allowed: 300  over for ms: 0.0190258026123047
12017/11/15 21:53:56.839:[WARN] Check timeout Node ip : 192.168.91.41 ,  THID 6 (taken: 300.019025802612   max_allowed: 300  over for ms: 0.0190258026123047
12017/11/15 21:53:56.839:[INFO]  Multi Thread execution done in :   300.23193359375(ms)
```

All the servers are checked in parallel, to reduce the execution time. 
But the thread execution has a timeout, to prevent the script to hang. 
The bad side of this is that if a node is not answering in time the information related to it will not fill and the node will be excluded. This is also tho prevent the script to consider a node that for any reason is not responsive.
The good side is that this timeout is configurable and you can easily identify what is the right value for your environment setting --check_timeout=800 (default 300) or even more, and see in the debug log how long it takes the script to execute the check. 

so in short:
set --check_timeout=800 in the scheduler and let me know
thanks.

**Comment by Tusamarco** [comment]
@altmannmarcelo  I had modified the check_timeout default and set to higher value. I realize that keeping it so low can be misleading at the start.

**Comment by altmannmarcelo** [comment]
@Tusamarco tested with latest version of script, I expected .41 to be promoted but .42 was promoted instead:

```
2017/11/16 15:46:05.883:[INFO] START EXECUTION
2017/11/16 15:46:05.883:[INFO]  Inizializing hostgroup 50 wwith maintenance HG 8050 bw
2017/11/16 15:46:05.883:[INFO]  Inizializing hostgroup 51 rwith maintenance HG 9051 mr
2017/11/16 15:46:05.903:[INFO]  Connecting to ProxySQL DBI:mysql:host=127.0.0.1;port=6032
2017/11/16 15:46:05.903:[INFO]  Galera cluster object created  
2017/11/16 15:46:05.906:[DEBUG] Calculate retry from comment Node:192.168.91.44 port:3306 hg:50 Time IN 
2017/11/16 15:46:05.906:[DEBUG] Calculate retry from comment Node:192.168.91.44 port:3306 hg:50 Time OUT 
2017/11/16 15:46:05.906:[INFO]  Galera cluster node   192.168.91.44:3306:HG=50
2017/11/16 15:46:05.906:[DEBUG] Calculate retry from comment Node:192.168.91.41 port:3306 hg:51 Time IN 
2017/11/16 15:46:05.906:[DEBUG] Calculate retry from comment Node:192.168.91.41 port:3306 hg:51 Time OUT 
2017/11/16 15:46:05.906:[INFO]  Galera cluster node   192.168.91.41:3306:HG=51
2017/11/16 15:46:05.906:[DEBUG] Calculate retry from comment Node:192.168.91.42 port:3306 hg:51 Time IN 
2017/11/16 15:46:05.906:[DEBUG] Calculate retry from comment Node:192.168.91.42 port:3306 hg:51 Time OUT 
2017/11/16 15:46:05.906:[INFO]  Galera cluster node   192.168.91.42:3306:HG=51
2017/11/16 15:46:05.906:[DEBUG] Calculate retry from comment Node:192.168.91.44 port:3306 hg:51 Time IN 
2017/11/16 15:46:05.907:[DEBUG] Calculate retry from comment Node:192.168.91.44 port:3306 hg:51 Time OUT 
2017/11/16 15:46:05.907:[INFO]  Galera cluster node   192.168.91.44:3306:HG=51
2017/11/16 15:46:05.907:[DEBUG] Calculate retry from comment Node:192.168.91.41 port:3306 hg:8050 Time IN 
2017/11/16 15:46:05.907:[DEBUG] Calculate retry from comment Node:192.168.91.41 port:3306 hg:8050 Time OUT 
2017/11/16 15:46:05.907:[INFO]  Galera cluster node   192.168.91.41:3306:HG=8050
2017/11/16 15:46:05.907:[DEBUG] Calculate retry from comment Node:192.168.91.42 port:3306 hg:8050 Time IN 
2017/11/16 15:46:05.907:[DEBUG] Calculate retry from comment Node:192.168.91.42 port:3306 hg:8050 Time OUT 
2017/11/16 15:46:05.907:[INFO]  Galera cluster node   192.168.91.42:3306:HG=8050
2017/11/16 15:46:05.907:[DEBUG] Calculate retry from comment Node:192.168.91.44 port:3306 hg:8050 Time IN 
2017/11/16 15:46:05.907:[DEBUG] Calculate retry from comment Node:192.168.91.44 port:3306 hg:8050 Time OUT 
2017/11/16 15:46:05.907:[INFO]  Galera cluster node   192.168.91.44:3306:HG=8050
2017/11/16 15:46:05.907:[INFO]  Galera cluster nodes loaded 
2017/11/16 15:46:05.907:[INFO]  Creating new thread to manage server check:192.168.91.44:3306:HG50
2017/11/16 15:46:05.923:[INFO]  Creating new thread to manage server check:192.168.91.41:3306:HG51
2017/11/16 15:46:05.939:[INFO]  Creating new thread to manage server check:192.168.91.42:3306:HG51
2017/11/16 15:46:05.954:[INFO]  Creating new thread to manage server check:192.168.91.44:3306:HG51
2017/11/16 15:46:05.939:[DEBUG]  Node check START 192.168.91.41:3306:HG51
2017/11/16 15:46:05.939:[DEBUG]  Getting connection START192.168.91.41:3306:HG51 
2017/11/16 15:46:05.941:[DEBUG]  Getting connection END192.168.91.41:3306:HG51 
2017/11/16 15:46:05.941:[DEBUG]  Getting NODE info START192.168.91.41:3306:HG51 
2017/11/16 15:46:05.957:[DEBUG]  Getting NODE info END192.168.91.41:3306:HG51 
2017/11/16 15:46:05.958:[DEBUG]  Node check END 192.168.91.41:3306:HG51
2017/11/16 15:46:05.969:[INFO]  Creating new thread to manage server check:192.168.91.41:3306:HG8050
2017/11/16 15:46:05.983:[INFO]  Creating new thread to manage server check:192.168.91.42:3306:HG8050
2017/11/16 15:46:05.954:[DEBUG]  Node check START 192.168.91.42:3306:HG51
2017/11/16 15:46:05.954:[DEBUG]  Getting connection START192.168.91.42:3306:HG51 
2017/11/16 15:46:05.960:[DEBUG]  Getting connection END192.168.91.42:3306:HG51 
2017/11/16 15:46:05.960:[DEBUG]  Getting NODE info START192.168.91.42:3306:HG51 
2017/11/16 15:46:05.986:[DEBUG]  Getting NODE info END192.168.91.42:3306:HG51 
2017/11/16 15:46:05.987:[DEBUG]  Node check END 192.168.91.42:3306:HG51
2017/11/16 15:46:05.998:[INFO]  Creating new thread to manage server check:192.168.91.44:3306:HG8050
2017/11/16 15:46:05.983:[DEBUG]  Node check START 192.168.91.41:3306:HG8050
2017/11/16 15:46:05.984:[DEBUG]  Getting connection START192.168.91.41:3306:HG8050 
2017/11/16 15:46:05.987:[DEBUG]  Getting connection END192.168.91.41:3306:HG8050 
2017/11/16 15:46:05.987:[DEBUG]  Getting NODE info START192.168.91.41:3306:HG8050 
2017/11/16 15:46:06.003:[DEBUG]  Getting NODE info END192.168.91.41:3306:HG8050 
2017/11/16 15:46:06.004:[DEBUG]  Node check END 192.168.91.41:3306:HG8050
2017/11/16 15:46:05.998:[DEBUG]  Node check START 192.168.91.42:3306:HG8050
2017/11/16 15:46:05.998:[DEBUG]  Getting connection START192.168.91.42:3306:HG8050 
2017/11/16 15:46:06.000:[DEBUG]  Getting connection END192.168.91.42:3306:HG8050 
2017/11/16 15:46:06.000:[DEBUG]  Getting NODE info START192.168.91.42:3306:HG8050 
2017/11/16 15:46:06.016:[DEBUG]  Getting NODE info END192.168.91.42:3306:HG8050 
2017/11/16 15:46:06.017:[DEBUG]  Node check END 192.168.91.42:3306:HG8050
2017/11/16 15:46:06.021:[INFO]  Thread joined :   2
2017/11/16 15:46:06.029:[INFO]  Thread joined :   3
2017/11/16 15:46:06.036:[INFO]  Thread joined :   5
2017/11/16 15:46:06.043:[INFO]  Thread joined :   6
2017/11/16 15:46:06.707:[WARN] Check timeout Node ip : 192.168.91.44 ,  THID 1 (taken: 800.004959106445   max_allowed: 800  over for ms: 0.0049591064453125  
12017/11/16 15:46:06.707:[WARN] Check timeout Node ip : 192.168.91.44 ,  THID 4 (taken: 800.004959106445   max_allowed: 800  over for ms: 0.0049591064453125  
12017/11/16 15:46:06.707:[WARN] Check timeout Node ip : 192.168.91.44 ,  THID 7 (taken: 800.004959106445   max_allowed: 800  over for ms: 0.0049591064453125  
12017/11/16 15:46:06.707:[INFO]  Multi Thread execution done in :   800.238132476807(ms) 
2017/11/16 15:46:06.707:[INFO]  Evaluate nodes state 
2017/11/16 15:46:06.708:[WARN] Fail-over in action Using Method = 1
2017/11/16 15:46:06.708:[INFO] Special Backup - Group found! I am electing a node to writer following the indications
 This Node Try to become the new WRITER for HG 50 Server details: 192.168.91.42:3306:HG8050
2017/11/16 15:46:06.708:[INFO] This Node Try to become a WRITER promoting to HG 50 192.168.91.42:3306:HG 8050
2017/11/16 15:46:06.722:[WARN]  Move node:192.168.91.42:330610001000 SQL:INSERT INTO mysql_servers (hostname,hostgroup_id,port,weight,max_connections) VALUES ('192.168.91.42',50,3306,1000,1000);
2017/11/16 15:46:06.722:[WARN] !!!! FAILOVER !!!!! 
 Cluster was without WRITER I have try to restore service promoting a node
2017/11/16 15:46:06.722:[INFO] END EXECUTION Total Time(ms):839.631080627441
2017/11/16 15:46:06.723:[INFO] 

```

**Comment by Tusamarco** [comment]
Ok this is a bug should be fix in https://github.com/Tusamarco/proxy_sql_tools/commit/1c8d4394c29c3fea0d783827c32da58feb12a96d

**Comment by Tusamarco** [comment]
@altmannmarcelo can we say THIS bug is fix and the other things are feature requests?

**Comment by altmannmarcelo** [comment]
@Tusamarco fix is working as expected. Thanks for the patch.
